proposals/lip-introduce_sidechain_recovery_mechanism.md

```diff
@@ -0,0 +1,555 @@
+                "dataType": "bytes"
+            },
+            "fieldNumber": 4
+        }
+    }
+}
+```
+
+#### State Recovery Command Verification
+
+Let `trs` be a transaction with module ID `MODULE_ID_INTEROPERABILITY` and command ID `COMMAND_ID_STATE_RECOVERY` to be verified. 
+Then `trs` is valid if the following logic returns `true`:
+
+```python
+sidechainAccount = terminatedAccount(trs.params.chainID)
+# the terminated account has to exist for this sidechain
+if sidechainAccount is empty:
+    return false
+
+let queryKeys and storeQueries be empty arrays
+
+for each entry in trs.params.storeEntries:
+    # the recover function corresponding to the module ID has to pass
+    route processing logic to the module given by trs.params.moduleID
+    if recover(trs.params.chainID, trs.params.moduleID, entry.storePrefix, entry.storekey, entry.storeValue) fails:
+        return false
+
+push entry.storeKey to queryKeys
+
+query = { key: entry.storeKey, 
+          value: SHA-256(entry.storeValue), 
+          bitmap: entry.bitmap}
+push query to storeQueries
+
+proofOfInclusionStores = { siblingHashes: trs.params.siblingHashes, queries : storeQueries}
+
+return SMTVerify(queryKeys, proofOfInclusionStores, sidechainAccount.stateRoot)
+```
+
+#### State Recovery Command Execution
+
+Processing a transaction `trs` with module ID `MODULE_ID_INTEROPERABILITY` and command ID `COMMAND_ID_STATE_RECOVERY` implies the following logic:
+
+```python
+sidechainAccount = terminatedAccount(trs.params.chainID)
+
+let storeQueries be an empty array
+
+for each entry in trs.params.storeEntries:
+    # the recover function corresponding to the module ID applies the recovery logic
+    route processing logic to the module given by trs.params.moduleID
+    recover(trs.params.chainID, trs.params.moduleID, entry.storePrefix, entry.storeKey, entry.storeValue)
+
+    emptyStore = empty bytes // define an empty store entry
+    query = { key: entry.storekey, 
+              value: SHA-256(emptyStore), 
+              bitmap: entry.bitmap}
+    push query to storeQueries
+
+sidechainAccount.stateRoot = SMTCalculateRoot(trs.params.siblingHashes, storeQueries)
+```
+#### Recover Function
+
+For the verification and application of this command it is assumed that the module given by `trs.params.moduleID` exposes a `recover` function, with the following interface:
+
+```
+recover(terminatedChainID, moduleID, storePrefix, storeKey, storeValue),
+```
+
+where:
+
+* `terminatedChainID`: The ID of the terminated chain.
+* `moduleID`: The ID of the recoverable module.
+* `storePrefix`: The store prefix of the store entry in the recoverable module state. 
+* `storeKey`: The store key of the store entry in the recoverable module state.
+* `storeValue`: The store value of the store entry in the recoverable module state.
+
+The recover function is specified for the [Token module][tokenReducer] and in the [NFT module][NFTReducer].
+
+### Recovery Initialization Command
```

Comment:
Here, could we align the name of the command with the command ID. So either calling the command `Initiate Recovery Command` or changing the ID to `COMMAND_ID_RECOVERY_INITIALIZATION`?

Reply:
I adapted the command ID as suggested.